File: frontend/src/types/project.ts
export type ProjectType = 'COMPETITION' | 'RESEARCH' | 'STARTUP' | 'OPENSOURCE' | 'OTHER'

export type ProjectStatus =
  | 'DRAFT'
  | 'RECRUITING'
  | 'IN_PROGRESS'
  | 'PENDING_REVIEW'
  | 'COMPLETED'
  | 'ARCHIVED'

export interface Project {
  id: number
  creatorId: number
  title: string
  projectType: ProjectType
  description?: string
  requirements?: string
  teamSize: number
  currentMembers: number
  expectedDuration?: number
  weeklyHours?: number
  expectedOutcome?: string
  status: ProjectStatus
  isRecommended: boolean
  views: number
  startDate?: string
  endDate?: string
  createdAt: string
  updatedAt: string
}

export interface ProjectSkillRequirement {
  id: number
  projectId: number
  skillName: string
  isRequired: boolean
  expectedLevel?: 'BEGINNER' | 'INTERMEDIATE' | 'EXPERT'
}

export interface ProjectMember {
  id: number
  projectId: number
  userId: number
  role?: string
  contribution?: string
  joinedAt: string
  leftAt?: string
  status: 'ACTIVE' | 'LEFT' | 'REMOVED'
}

export interface ProjectApplication {
  id: number
  projectId: number
  applicantId: number
  applicationReason?: string
  status: 'PENDING' | 'APPROVED' | 'REJECTED' | 'WITHDRAWN'
  reviewedBy?: number
  reviewComment?: string
  appliedAt: string
  reviewedAt?: string

  // VO 字段（后端 ApplicationVO 返回）
  projectTitle?: string
  applicantName?: string
  applicantAvatar?: string
  applicantSkills?: string[]
}

export interface ProjectComment {
  id: number
  projectId: number
  parentId?: number
  replyToUserId?: number
  userId: number
  content: string
  likeCount?: number
  createdAt: string

  // 用户信息
  username?: string
  nickname?: string
  avatar?: string

  // 回复目标用户信息
  replyToUsername?: string
  replyToNickname?: string

  // 子评论
  replies?: ProjectComment[]
}

export interface ProjectFile {
  id: number
  projectId: number
  uploaderId: number
  fileName: string
  originalFileName?: string
  fileUrl: string
  fileType: 'DOCUMENT' | 'IMAGE' | 'CODE' | 'OTHER'
  category: string
  fileSize: number
  mimeType?: string
  description?: string
  createdAt: string
  
  // 上传者信息
  uploaderName?: string
  uploaderAvatar?: string
}

export interface ProjectMilestone {
  id: number
  projectId: number
  title: string
  status: 'PLANNED' | 'IN_PROGRESS' | 'DONE'
  plannedAt?: string
  actualAt?: string
  ownerId?: number
  ownerName?: string
  remark?: string
  sortOrder?: number
  createdAt?: string
  updatedAt?: string
}

File: backend/teamup-server/src/main/java/com/teamup/server/modules/project/entity/Project.java
package com.teamup.server.modules.project.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.TableField;
import lombok.Data;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 项目实体
 */
@Data
@TableName("projects")
public class Project {
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private Long creatorId;
    
    @TableField(exist = false)  // 不映射到数据库字段
    private String creatorName;  // 创建者用户名（用于前端显示）
    
    private String title;
    private String projectType;  // COMPETITION, RESEARCH, STARTUP, OPENSOURCE, OTHER
    private String description;
    private String requirements;
    private Integer teamSize;
    private Integer currentMembers;
    private Integer expectedDuration;
    private Integer weeklyHours;
    private String expectedOutcome;
    private String status;  // DRAFT, RECRUITING, IN_PROGRESS, PENDING_REVIEW, COMPLETED, ARCHIVED
    private Boolean isRecommended;
    private Integer views;
    private LocalDate startDate;
    private LocalDate endDate;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private Long competitionId;  // 关联的比赛ID（可选）
    private Long teamId;  // 执行团队ID（招募完成后生成）
    private String teamMode;  // 团队模式：CREATE_NEW-创建新团队, USE_EXISTING-使用已有团队
}

